FilterPlugin/src/main/java/io/antmedia/filter/FilterAdaptor.java

```diff
@@ -333,70 +332,54 @@ public boolean update()
 					}
 				}
 			});
-
-			if(prevAudioFilterGraph != null) {
-				prevAudioFilterGraph.close();
-			}
 		}
 		
-		return true;
+		return result;
 	}
 	
 	/*
 	 * This method is used for the creation and also for the update of the filter
 	 * For example new inputs mat be added as an update
 	 */
 	
-	public synchronized boolean createFilter(FilterConfiguration filterConfiguration, AntMediaApplicationAdapter app) {
+	public synchronized Result startFilterProcess(FilterConfiguration filterConfiguration, AntMediaApplicationAdapter app) {
 		if(vertx == null) {
 			vertx = app.getVertx();
 		}
 		
+		Result result = new Result(true);
+		
 		this.filterConfiguration = filterConfiguration;
 		
+		// Check database if stream is not exist or stream is not broadcasting status
+   		for(String streamId : filterConfiguration.getInputStreams()) {
+   			Broadcast broadcast = app.getDataStore().get(streamId);
+   			if(broadcast == null || (broadcast != null && !broadcast.getStatus().contains(IAntMediaStreamHandler.BROADCAST_STATUS_BROADCASTING))) {
```

Comment:
Generally this approach is better to check if it equals for `nullpointerexception`
```
IAntMediaStreamHandler.BROADCAST_STATUS_BROADCASTING.equals(broadcast.getStatus)
````